### PR TITLE
Partial search

### PR DESCRIPTION
**Pull request by polygirl**
Fixes issue #215

### CODE REVIEW

File: src/database/series.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ function SeriesVolume(seriesId: string, volume: string, startYear: number, title
   if (Number(volume) > 1) {
     this.titleWithVolume += ' Vol. ' + volume;
   }
+  this.searchTitleWithVolume = title += ' Vol. ' + volume;
```
Comment (SubJunk) [inline src/database/series.ts]: What does the `+=` here do?

File: src/app/filter.pipe.ts [hunk 2]
```diff
@@ -12,9 +12,15 @@ export class FilterPipe implements PipeTransform {
...
 
     searchText = searchText.toLowerCase();
+    // split the search string into an array of individual words
+    const searchTextAsWords = searchText.match(/\s+/g) || [];
 
     return items.filter( item => {
-      return item.displayText.toLowerCase().includes(searchText);
+      // results should contain all search words
+      return searchTextAsWords.every(matchAll);
+      function matchAll(wordItem) {
```
Comment (SubJunk) [inline src/app/filter.pipe.ts]: It might be more readable to have this be an inline, anonymous function

File: e2e/src/search.e2e-spec.ts [hunk 3]
```diff
@@ -16,11 +16,11 @@ describe('Search', () => {
     'should display the search div\n' +
     'should display correct search results',
   async () => {
-    await element(by.tagName('input')).sendKeys('Uncanny X-Men #100');
+    await element(by.tagName('input')).sendKeys('Uncanny X-Men Vol. 1 #1');
     expect(
       await element(by.css('.search-result')).getText()
     ).toContain(
-      'Uncanny X-Men #100'
+      'Uncanny X-Men Vol. 1 #1'
```
Comment (SubJunk) [inline e2e/src/search.e2e-spec.ts]: It would be good to add testing for the new search functionality in here

File: src/database/series.ts (hunk 1 above)
Comment (polygirl) [reply src/database/series.ts]: Copy paste error and  not for this branch

File: src/app/filter.pipe.ts (hunk 2 above)
Comment (polygirl) [reply src/app/filter.pipe.ts]: I'm sure that's true.  I don't know how to do that.

**Pull request closed without merging**